### PR TITLE
Add safeguards around imagestreams with no tags

### PR DESCRIPTION
**Pull request by anishasthana**
In situations where the imagestream has no available tags the value for
imagestream_name never gets set, resulting in the build_status
assignment failing. This change will result in "latest" being used as
the default tag.

Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

### CODE REVIEW

File: jupyterhub_singleuser_profiles/images.py [hunk 1]
```diff
@@ -100,7 +103,7 @@ def load(self):
                     tag_annotations = tag.annotations
 
                 # Default name if there is no display name annotation
```
Comment (vpavlin) [inline jupyterhub_singleuser_profiles/images.py:105]: I think this line has to move with the `imagestream_name` definition

**Comment by vpavlin** [comment]
I am not sure I fully understand this. Any imagestream we expect to show has to have tags, otherwise it would could not have the tag annotations. 

I think that the right way to solve this would be to ignore imagestreams without tags, but print our an error.

Can you share what issue lead to this PR?

**Comment by mroman-redhat** [comment]
@vpavlin We encountered this error on several clusters actually, where jsp would error out when assigning build_status. There is no issue assigned to it in jira.

**Comment by anishasthana** [comment]
I'm good with ignoring tagless imagestreams, I'll try updating this PR accordingly.

File: jupyterhub_singleuser_profiles/images.py [hunk 2]
```diff
@@ -107,17 +110,17 @@ def load(self):
...
                                             name=tag.name,
-                                            recommended=tag_annotations.get(RECOMMENDED_ANNOTATION, False)
+                                            recommended=tag_annotations.get(RECOMMENDED_ANNOTATION, False),
+                                            build_status=self.get_image_build_status(imagestream_name)
                 ))
 
             result.append(ImageInfo(description=annotations.get(DESCRIPTION_ANNOTATION),
                                 url=annotations.get(URL_ANNOTATION),
-                                display_name=annotations.get(DISPLAY_NAME_ANNOTATION) or imagestream_name,
+                                display_name=annotations.get(DISPLAY_NAME_ANNOTATION) or i.metadata.name,
```
Comment (vpavlin) [inline jupyterhub_singleuser_profiles/images.py:119]: You can use `imagestram_name` here safely now

**Comment by vpavlin** [comment]
lgtm

**Pull request merged**